lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -103,6 +118,40 @@ async def fetch_form(
                     files2.append(value)
         return self.service.create_fetch(trans, payload, files2)
 
+    @router.post(
+        "/api/tools",
+        summary="Execute tool with a given parameter payload",
+        route_class_override=JsonApiRoute,
+    )
+    async def execute_json(
+        self,
+        payload: CreateToolBody,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+    ) -> ToolResponse:
+        return self.service.execute(trans, payload)
+
+    @router.post(
+        "/api/tools",
+        summary="Execute tool with a given parameter payload",
+        route_class_override=FormDataApiRoute,
+    )
+    async def execute_form(
```

Comment:
Those are definitely not async as they interact with the database, what's your reasoning here ?

Reply:
I see, I've made that mistake myself 😆

Reply:
I changed it.